### PR TITLE
Ensure order association is set correctly when adding admin adjustments

### LINKED ISSUE #7108: Order Adjustments added manually (Edit Order) not reflecting on customer total
## Description

Hub wishes to issue a partial refund/increase for a product (ie. not refund/double the full cost of that product but just part of it).  
Hub manager visits Edit Order -> Adjustments (https://openfoodnetwork.org.uk/admin/orders/Rxxxxxxxx/adjustments) and creates a negative/positive adjustment.
After adjustment has been created the customer's total balance should be updated accordingly.  This is not the case.
Result: 
Managers can only edit their orders by adding or removing full products: non ideal!!!
Means they are unable to issue/record refunds or extra monies collected.

## Expected Behavior

Add an adjustment to an order https://openfoodnetwork.org.uk/admin/orders/Rxxxxxxxx/adjustments.
Click 'Continue'
Order balance updated accordingly.

Click on 'edit' icon next to order adjustment and can edit text and value of the adjustment.

## Actual Behaviour

Add an adjustment to an order https://openfoodnetwork.org.uk/admin/orders/Rxxxxxxxx/adjustments.
Click 'Continue'
Order balance remains unchanged.

Click on 'edit' icon next to order adjustment and adjustment is copied.  No access to Edit.

## Steps to Reproduce

1. Create an order and pay for it.
2. Edit order 
3. Create a negative adjustment to the order on page https://openfoodnetwork.org.uk/admin/orders/Rxxxxxxxx/adjustments
4. Observe balance remains unchanged
5. Create a positive adjustment to the order on the same page
6. Observe balance remains unchanged
7. Click on 'Edit' adjustment icon
8. observe this copies the adjustment but no access to edit

## Severity

S2

## Your Environment

* Version used:
* Browser name and version:
* Operating System and version (desktop or mobile):

### PR DESCRIPTION
**Pull request by Matt-Yorkley**
#### What? Why?

Closes  #7108

The adjustment's order_id was not being set when creating admin adjustments via the UI, which meant some of the scopes were not applied correctly in some cases.

#### What should we test?

When creating an adjustment via admin, the adjustment is added to the order total when recalculated.

#### Release notes

Fixed an issue with creating admin adjustments

Changelog Category: User facing changes | Technical changes

### CODE REVIEW

**Review by sauloperez — APPROVED** [review]

**Comment by Matt-Yorkley** [comment]
Added a commit to address: https://github.com/openfoodfoundation/openfoodnetwork/issues/7108#issuecomment-799347430

File: app/controllers/spree/admin/adjustments_controller.rb [hunk 1]
```diff
@@ -2,16 +2,18 @@ module Spree
   module Admin
     class AdjustmentsController < ::Admin::ResourceController
       belongs_to 'spree/order', find_by: :number
-      destroy.after :reload_order
 
       prepend_before_action :set_included_tax, only: [:create, :update]
+      before_action :set_order_id, only: [:create, :update]
+      after_action :update_order, only: [:create, :update, :destroy]
```
Comment (sauloperez) [inline app/controllers/spree/admin/adjustments_controller.rb:8]: this is getting a bit crazy IMO. I think it's time we implement these action methods and call `super` when needed. I would do it in a separate PR once we know this fixes the issue.

**Review by sauloperez — APPROVED** [review]

File: app/controllers/spree/admin/adjustments_controller.rb (hunk 1 above)
Comment (Matt-Yorkley) [reply app/controllers/spree/admin/adjustments_controller.rb:8]: Agreed, this whole controller needs some serious work.

**Comment by filipefurtad0** [comment]
Hey @Matt-Yorkley,

**Before staging this PR**, following the steps to reproduce on issue #7108. For an order in state 'complete', editing the order and performing the following operations on the Adjustments tab:

- creating an Adjustment
The adjustment is created, but the value is not added up to the order total (and hence, does not add up to the respective customer balance -> /customers). Clicking "Update and Recalculate fees" makes no changes.

- Deleting/Editing the adjustment
Deletion: Makes no changes - appears successful, but refreshing the page shows makes the adjustment re-appear
Editing: Takes no effect -> error 304:

image


**After this PR**

- creating an Adjustment - OK
Adjustments amounts now add up to the order total, and hence appear on the customer balance.

- Editing an adjustment - OK
This works now. Editing the i) name, ii) amounts, iii) tax-settings now work fine.

The exception is, those adjustments created before this PR; i.e., probably, those created after the regression was introduced: these adjustments cannot seem to be edited, or deleted. However, they do not contribute to the order value (which is the original issue):

Deleting, and attempting to edit the "just because" adjustment:
 
Peek 2021-03-15 15-38


Please note that after this PR I could not reproduce issue #5422 :tada: 

Peek 2021-03-15 15-28

This PR fixes the issue and is good to go :tada: 

Do you think we need an issue to deal with previously created adjustments @Matt-Yorkley ?

**Comment by Matt-Yorkley** [comment]
> Do you think we need an issue to deal with previously created adjustments

Yeah I think there could be a few of those adjustments that can't be edited. I think we can make a data migration to deal with them, but we can handle it separately.

**Comment by filipefurtad0** [comment]
Created this issue to deal with the order adjustments created recently, between the regression and the deployment of the fix:
https://github.com/openfoodfoundation/openfoodnetwork/issues/7120

**Pull request merged**